rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend

```diff
@@ -92,45 +104,96 @@ class TclLauncherUi implements Launcher {
 			project.refreshLocal(IProject.DEPTH_INFINITE, monitor)
-			if (result.expectedFile == null) {
+			if (result.expectedFileRoot == null) {
 				logger.error("resulting expectedFile must not be null")
 			} else {
-				safeUpdateJunitTestView(elementId, result.expectedFile)
+				safeUpdateJunitTestView(elementId, result.expectedFileRoot, project.name)
 			}
 			monitor.done
 		])
 		return true
 	}
 
+	def String createTestCasesCommaList(IStructuredSelection selection) {
+		if (launchShortcutUtil == null) {
+			launchShortcutUtil = tclInjectorProvider.get.getInstance(LaunchShortcutUtil)
+		}
+		if (selection.size > 1) {
+			return selection.toList.map[launchShortcutUtil.getQualifiedNameForTestInTcl(it as IResource).toString].
+				join(",")
+		} else {
+			return launchShortcutUtil.getQualifiedNameForTestInTcl(selection.firstElement as IResource).toString
+		}
+	}
+
 	/** 
 	 * provide test result file (either the one created by the test run, or, if absent/on error 
 	 * a default error file) that is imported into junit (and thus displayed) 
 	 * */
-	private def void safeUpdateJunitTestView(String elementId, File expectedFile) {
-		val parentFolder = new File(expectedFile.parent)
-		if (!parentFolder.exists && !parentFolder.mkdirs) {
-			logger.error("failed to change into parentFolder='{}'", parentFolder.absolutePath)
+	private def void safeUpdateJunitTestView(String elementId, File expectedFile, String projectName) {
+		logger.debug("Test result parent dir {}", expectedFile)
+		val xmlResults = expectedFile.listFiles(new FileFilter() {
+
+			override accept(File pathname) {
+				return pathname.isFile && pathname.name.endsWith(".xml")
+			}
+
+		})
+		val resultFile = new File(expectedFile, "te-testCompose.xml")
+		if (xmlResults.length > 0) {
+			writeTestResultFile(projectName, resultFile, xmlResults)
 		} else {
-			val newFile = File.createTempFile(expectedFile.name, ".xml", parentFolder)
-			if (expectedFile.exists) {
-				try {
-					Files.move(expectedFile, newFile)
-				} catch (IOException e) {
-					logger.error('''error during storage of test run result expectedFile='«expectedFile.path»' ''', e)
-					writeErrorFile(elementId, newFile)
+			writeErrorFile(elementId, resultFile)
+		}
+		JUnitCore.importTestRunSession(resultFile)
+	}
+
+	def void writeTestResultFile(String projectName, File resultFile, File[] xmlResults) {
+		val docBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder
+		val resultDoc = docBuilder.newDocument
+		val testRun = resultDoc.createElement("testrun")
+		resultDoc.appendChild(testRun)
+		var testCount = 0;
+		var failureCount = 0;
+		var errorsCount = 0;
+		var ignoreCount = 0;
+		for (file : xmlResults) {
+			val suiteDoc = docBuilder.parse(file)
+			val noteList = suiteDoc.childNodes
+			for (var i = 0; i < noteList.length; i++) {
+				if (noteList.item(i).nodeName.equals("testsuite")) {
+					testRun.appendChild(resultDoc.importNode(noteList.item(i), true))
+					testCount = testCount +
```

Comment:
is changed, but i think the long term is a little bit more clear. Maybe I have to learn reading this short syntax.